Added onSwiping method

Purii opened the pull request:
Great component!

I needed a method to get notified, once the user started swiping the modal:
Modal from bottom containing an arrow icon. The icon shall be animated based on the swiping state.

Feel free to provide improvements 👍 

<img width="401" alt="example" src="https://user-images.githubusercontent.com/7056989/52448278-b31f2500-2b33-11e9-8329-d51d8a5e55df.png">

mmazzarolo commented:
@Purii thank you for the PR!
You're right, a few more "swiping" props would be useful. 
My main issue with the PR is the naming convention:
- `onSwipe`
- `onSwiping`
- `onSwipingDone`
They seem a bit confusing now, but that's just because the original `onSwipe` props should have been named `onSwipeDone`/`onSwipeComplete` instead... 🤔 

Would it make sense doing something like this (just asking for feedback, I'm not saying you should do it in the PR 😃)?
- Deprecate the `onSwipe` callback (but still support it)
- Create the following callbacks:
  * `onSwipeStart`
  * `onSwipeMove`
  * `onSwipeComplete`
  * `onSwipeCancel`

What do you think?

Purii commented:
I'm also not that happy with the naming.
Your suggestions sounds much better. I'm going to update the PR :)

mmazzarolo commented:
@Purii thank you sir 🙇

Purii commented:
@mmazzarolo You're welcome!
I added the four new methods.

**Deprecation `onSwipe`**
The deprecation warning will be triggered on `componentDidMount`.

**Added Prettier Config**
During development I saw, that my local prettier config is different than the one used in the repo.
So I added a prettier config to override different local settings.
I also set `trailingComma:true`.
Hope this is fine for you?

Let me know if I should change something :)

Purii commented:
Any idea for a better naming of `percentageLeft` as value of `onSwipeMove`?

mmazzarolo commented on src/index.js:
```diff
@@ -148,25 +151,31 @@ class ReactNativeModal extends Component {
+      console.warn(
+        '`<Modal onSwipe="..." />` is deprecated. Use `<Modal onSwipeComplete="..." />` instead.',
+      );
+    }
```
❤️

mmazzarolo commented on src/index.js:
```diff
@@ -201,16 +210,20 @@ class ReactNativeModal extends Component {
-          }
+        if (this.props.scrollTo && this.props.scrollOffset > 0) {
+          return false; // user needs to be able to scroll content back up
+        }
```
Thanks for the cleanup

Purii commented:
> `swipeProgress` maybe? Did you check if it is invoked correctly? (Can't check it right now).

Yep. Will be invoked on each move. No throttling, as I would leave that responsibility to the user.
Currently I just return `newOpacityFactor`, that's why I labeled it `percentageLeft`. Wouldn't `swipeProgress` implicate the progress of the swipe and not the way still to go? But we could give it a try: `swipeProgress === newOpacityFactor`

mmazzarolo commented:
> Wouldn't `swipeProgress` implicate the progress of the swipe and not the way still to go?

🤦‍♂️ my bad, you're right. 
Mind reverting it for the last time? Thank you!

Purii commented:
Renamed it to `percentageShown`. Still not perfect, but understandable.
Also updated the example: `onSwipe` --> `onSwipeComplete` 👍

Purii commented:
Is the prettier config fine for you?
Changed some formatting of the example (added trailing comma,...)

mmazzarolo commented:
> Is the prettier config fine for you?

Yes, see:  
> No worries about the prettier config, we are planning to start using a `community` driven one so it probably be changed again soon.


Merging, thanks @Purii !

mmazzarolo commented:
Landed in [v8.0.0-beta.1](https://github.com/react-native-community/react-native-modal/releases/tag/v8.0.0-beta.1) 🛬
